### PR TITLE
Support avro-schema references for records

### LINKED ISSUE #116: Add support for type references by name in avro-schemas
```
{
  "name": "users",
  "type": "record",
  "fields": [
      {
        "name": "User_A",
        "type": {
	    "name": "user",
            "type": "record",
            "fields": [
              {"name": "username", "type": "string"},
              {"name": "phone", "type": "long"},
              {"name": "age", "type": "int"}
            ]
        }
      },
      {"name": "User_B", "type": "user"}
    ]
}
```

### PR DESCRIPTION
**Pull request by Totktonada**
Fixes #116.

### CODE REVIEW

File: graphql/tarantool_graphql.lua [hunk 1]
```diff
@@ -949,6 +970,8 @@ gql_type = function(state, avro_schema, collection, collection_name, field_name)
                 avro_schema.name,
             fields = fields,
         })
+        state.declarations[avro_schema.name] = types.nonNull(res)
```
Comment (SudoBobo) [inline graphql/tarantool_graphql.lua]: As far as I understand we do not check anywhere if names of named types in our schema are unique (in context of type defining, not referencing) Avro-schema spec states: "A schema or protocol may not contain multiple definitions of a fullname" Moreover on line 973 we may  have over writing, haven't we? As in our typical case we have quite massive schemas, unconscious type re-definition may be a problem. So my idea to add some assertion for type re-defenition.

File: graphql/tarantool_graphql.lua [hunk 2]
```diff
@@ -980,6 +1003,12 @@ gql_type = function(state, avro_schema, collection, collection_name, field_name)
     elseif avro_t == 'union' then
         return create_gql_union(state, avro_schema, field_name)
     else
+        if type(avro_schema) == 'string' then
+            if state.declarations[avro_schema] ~= nil then
+                return state.declarations[avro_schema]
+            end
+        end
+
```
Comment (SudoBobo) [inline graphql/tarantool_graphql.lua]: Do I understand right that order of definitions in our avro-schema matters? As far as I understand now if we try to put type reference BEFORE type definition in our avro-schema we will have 'unrecognized avro-schema type'. Maybe it is worthy to make additional check and return something like 'reference of yet undefined avro-schema type'

File: graphql/tarantool_graphql.lua (hunk 1 above)
Comment (Totktonada) [reply graphql/tarantool_graphql.lua]: It is definitions, sorry. Will rename. Assert seems to make sense for proper error message, will add.

File: graphql/tarantool_graphql.lua (hunk 2 above)
Comment (Totktonada) [reply graphql/tarantool_graphql.lua]: Order is matters. You cited the standard right above :)

'unrecognized avro-schema type' seems ok for me. I’ll see whether adding additional check will looks good.

Comment (Totktonada) [reply graphql/tarantool_graphql.lua]: We fall into this error in the two cases:

1. Forward reference (forbidden by the standard).
2. Non-defined type.

I don’t want to traverse all schema to give some-order-better error message. Will leave as is.

**Comment by Totktonada** [comment]
@SudoBobo Force-pushed.

**Review by SudoBobo — APPROVED** [review]

**Pull request merged**